src/tests/ftest/control/dmg_system_reformat.py

```diff
@@ -0,0 +1,129 @@
+                i.e. 0.5 (50%) 0.25(25%)
+
+        Raises:
+            CommandFailure: raise exception if pool creation fails.
+
+        Returns:
+            TestPool: the created test pool object.
+
+        """
+        if percentage > 1 or percentage <= 0:
+            self.fail("The percent value provided cannot be used: %s percent",
+                      percentage * 100)
+        # Convert info from dmg's human readable to bytes
+        scm_size = int(percentage * human_to_bytes(self.scm_cap))
+
+        # Update size value with unit and create, create pool object.
+        self.log.info("Create pool at %s percent capacity.", percentage * 100)
+        pool = self.get_pool(create=False, connect=False)
+        pool.scm_size.update(scm_size)
+        pool.create()
+
+        return pool
+
+    @fail_on(ServerFailed)
+    def test_dmg_system_reformat(self):
+        """
+        JIRA ID: DAOS-5415
+
+        Test Description: Test dmg system reformat functionality.
+
+        :avocado: tags=all,small,pr,hw,control,sys_reformat,dmg
+        """
+        self.create_pool_at_capacity(0.7)
+
+        self.log.info("Check that new pool will fail with DER_NOSPACE")
+        try:
+            pool = self.create_pool_at_capacity(0.7)
+        except TestFail as err:
+            self.log.info("Pool create failed: %s", str(err))
+            if pool.dmg.result.stderr:
+                if "DER_NOSPACE" not in pool.dmg.result.stderr:
+                    self.fail("Pool create did not fail do to DER_NOSPACE!")
+
+        self.log.info("Stop running io_server instancess: 'dmg system stop'")
+        self.get_dmg_command().system_stop(force=True)
+        if self.get_dmg_command().result.exit_status != 0:
+            self.fail("Detected issues performing a system stop: {}".format(
+                self.get_dmg_command().result.stderr))
+
+        self.log.info("Perform dmg storage format on all system ranks:")
+        self.get_dmg_command().storage_format(reformat=True)
+        if self.get_dmg_command().result.exit_status != 0:
+            self.fail("Detected issues performing storage format: {}".format(
+                self.get_dmg_command().result.stderr))
+
+        # Check that io_servers starts up again
+        self.log.info("<SERVER> Waiting for the daos_io_servers to start")
+        self.server_managers[-1].manager.job.pattern_count = 2
+        if not self.server_managers[-1].manager.job.\
+           check_subprocess_status(self.server_managers[-1].manager.process):
+            self.server_managers[-1].kill()
+            raise ServerFailed("Failed to start servers after format")
+
+        # Check that we have cleared storage by checking pool list
+        if self.get_dmg_command().pool_list():
+            self.fail("Detected pools in storage after refomat: {}".format(
+                self.get_dmg_command().result.stdout))
+
+        # Create last pool now that memory has been wiped.
+        self.create_pool_at_capacity(0.7)
```

Comment:
Don't we need try-except here like we have in line 95 to 97?

Reply:
Yes, you are right. I'll add